main/src/main/java/org/mobilitydata/gtfsvalidator/table/GeoJsonFileLoader.java

```diff
@@ -0,0 +1,224 @@
+                  + '.'
+                  + GtfsGeoJsonFeature.GEOMETRY_TYPE_FIELD_NAME);
+        } else if (!geometry.has(GtfsGeoJsonFeature.GEOMETRY_COORDINATES_FIELD_NAME)) {
+          missingRequiredFields.add(
+              GtfsGeoJsonFeature.FEATURE_COLLECTION_FIELD_NAME
+                  + '.'
+                  + GtfsGeoJsonFeature.GEOMETRY_FIELD_NAME
+                  + '.'
+                  + GtfsGeoJsonFeature.GEOMETRY_COORDINATES_FIELD_NAME);
+        } else if (missingRequiredFields
+            .isEmpty()) { // All required fields are present - Validate geometry
+          // Create a new GtfsGeoJsonFeature
+          gtfsGeoJsonFeature = new GtfsGeoJsonFeature();
+          gtfsGeoJsonFeature.setFeatureId(
+              featureObject.get(GtfsGeoJsonFeature.FEATURE_ID_FIELD_NAME).getAsString());
+
+          String type = geometry.get(GtfsGeoJsonFeature.GEOMETRY_TYPE_FIELD_NAME).getAsString();
+
+          if (type.equals(GeometryType.POLYGON.getType())) {
+            gtfsGeoJsonFeature.setGeometryType(GeometryType.POLYGON);
+            Polygon polygon =
+                geometryValidator.createPolygon(
+                    geometry.getAsJsonArray(GtfsGeoJsonFeature.GEOMETRY_COORDINATES_FIELD_NAME),
+                    gtfsGeoJsonFeature);
+            if (polygon == null) return null;
+            gtfsGeoJsonFeature.setGeometryDefinition(polygon);
+
+          } else if (type.equals(GeometryType.MULTI_POLYGON.getType())) {
+            gtfsGeoJsonFeature.setGeometryType(GeometryType.MULTI_POLYGON);
+            MultiPolygon multiPolygon =
+                geometryValidator.createMultiPolygon(
+                    geometry.getAsJsonArray(GtfsGeoJsonFeature.GEOMETRY_COORDINATES_FIELD_NAME),
+                    gtfsGeoJsonFeature);
+            if (multiPolygon == null) return null;
+            gtfsGeoJsonFeature.setGeometryDefinition(multiPolygon);
+
+          } else {
+            noticeContainer.addValidationNotice(
+                new UnsupportedGeometryTypeNotice(featureIndex, featureId, type));
+          }
+          JsonObject properties =
+              featureObject.getAsJsonObject(GtfsGeoJsonFeature.FEATURE_PROPERTIES_FIELD_NAME);
+          if (properties.has(GtfsGeoJsonFeature.FEATURE_PROPERTIES_STOP_NAME_FIELD_NAME)) {
+            gtfsGeoJsonFeature.setStopName(
+                properties
+                    .get(GtfsGeoJsonFeature.FEATURE_PROPERTIES_STOP_NAME_FIELD_NAME)
+                    .getAsString());
+          }
+          if (properties.has(GtfsGeoJsonFeature.FEATURE_PROPERTIES_STOP_DESC_FIELD_NAME)) {
+            gtfsGeoJsonFeature.setStopDesc(
+                properties
+                    .get(GtfsGeoJsonFeature.FEATURE_PROPERTIES_STOP_DESC_FIELD_NAME)
+                    .getAsString());
+          }
+
+          return gtfsGeoJsonFeature;
+        }
+      }
+    }
+    addMissingRequiredFieldsNotice(missingRequiredFields, noticeContainer, featureId, featureIndex);
```

Comment:
Maybe plural, since we potentially add multiple notices
```suggestion
    addMissingRequiredFieldsNotices(missingRequiredFields, noticeContainer, featureId, featureIndex);
```